### PR TITLE
Add guild folders

### PR DESCRIPTION
**Pull request by diamondburned**

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -694,13 +694,22 @@ type Settings struct {
...
 	FriendSourceFlags      *FriendSourceFlags `json:"friend_source_flags"`
 	Status                 Status             `json:"status"`
 	DetectPlatformAccounts bool               `json:"detect_platform_accounts"`
 	DeveloperMode          bool               `json:"developer_mode"`
 }
 
+// GuildFolder holds a single folder that you see in the left guild panel.
+type GuildFolder struct {
+	Name     string   `json:"name"`
+	ID       int64    `json:"id"`
```
Comment (ewohltman) [inline structs.go]: In other structs, the `ID` field is a string. Should this also be a string instead of an int64?

**Comment by Bios-Marcel** [comment]
Is there no eventing or anything for this? What happens if a guild folder gets deleted or changed in another instance? Will the cache be inconsistent?

File: structs.go (hunk 1 above)
Comment (diamondburned) [reply structs.go]: This isn't a snowflake, it's rather just arbitrary numbers (e.g. `3185939939`)

**Comment by diamondburned** [comment]
> Is there no eventing or anything for this? What happens if a guild folder gets deleted or changed in another instance? Will the cache be inconsistent?

Fair point, I'll add those

**Pull request closed without merging**